MAINTAINERS.md

```diff
@@ -1,4 +1,30 @@
-# Release process
+# Maintainers guide
+
+This document is intended for _maintainers_. Anyone with commit access to the Toolpad respository.
+
+## Pull request labels
+
+Each pull request must have a label applied. We apply labels in the following dimensions:
+
+### location
+
+- `docs`: For updates to the `./docs` folder.
+- `core`: For updates Toolpad application and its build environment.
```

Comment:
This is not how core is documented in https://www.notion.so/mui-org/core-c4efa9cabeab49248ff39e583807a2aa. For example, I don't think that https://github.com/mui/mui-toolpad/pull/1218 is about `core` but it directly impacts users, but https://github.com/mui/mui-toolpad/pull/1211 is about `core` because users won't directly care.